src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -16,6 +16,33 @@ public Rectangle(Integer width, Integer length, String color) {
         this.color = color;
     }
 
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        Rectangle rectangle = (Rectangle) o;
+        if (rectangle.color == null || rectangle.width == null || rectangle.length == null
+                || color == null || width == null || length == null) {
+            return this.hashCode() == rectangle.hashCode();
+        }
+        return width.equals(rectangle.width) && length.equals(rectangle.length)
+                && color.equals(rectangle.color);
+    }
+
+    @Override
+    public int hashCode() {
+        int primecode = 31;
+        int result = 1;
+        result = primecode * result + (width == null ? 0 : width);
+        result = primecode * result + (length == null ? 0 : length);
```

Comment:
see checklist: you need to use hashcodes of fields

Reply:
thought that need hashcodes only in non-numeric fields
fixed